### PR TITLE
probe-app: add support for non-audio probe buffers

### PR DESCRIPTION
**Pull request by kv2019i**
Series of patches to add non-audio probe buffer support (and few minor fixes).

### CODE REVIEW

**Comment by kv2019i** [comment]
Converting to a draft while debugging a possible bug.

File: tools/probes/probes_main.c [hunk 1]
```diff
@@ -100,9 +101,18 @@ int get_buffer_file_free(struct wave_files *files)
...
+	if ((format & PROBE_MASK_FMT_TYPE) && (format & PROBE_MASK_AUDIO_FMT) == 0)
+		return true;
+
+	return false;
+}
+
 int init_wave(struct wave_files *files, uint32_t buffer_id, uint32_t format)
 {
 	char path[FILE_PATH_LIMIT];
+	bool audio = is_audio_format(format);
```
Comment (ujfalusi) [inline tools/probes/probes_main.c:115]: nitpick: move one line up?

File: tools/probes/probes_main.c [hunk 2]
```diff
@@ -111,10 +121,13 @@ int init_wave(struct wave_files *files, uint32_t buffer_id, uint32_t format)
...
-	fprintf(stdout, "%s:\t Creating wave file for buffer id: %d\n",
-		APP_NAME, buffer_id);
+	fprintf(stdout, "%s:\t Creating %s file for buffer id: %d\n",
+		APP_NAME, audio ? "wave" : "raw", buffer_id);
 
-	sprintf(path, "buffer_%d.wav", buffer_id);
+	if (audio)
+		sprintf(path, "buffer_%d.wav", buffer_id);
+	else
+		sprintf(path, "buffer_%d.raw", buffer_id);
```
Comment (ujfalusi) [inline tools/probes/probes_main.c:130]: nitpick: `sprintf(path, "buffer_%d.%s", buffer_id, audio ? "wave" : "raw");`

but your version is likely more readable..

**Review by ujfalusi — APPROVED** [review]

File: tools/probes/probes_main.c (hunk 1 above)
Comment (kv2019i) [reply tools/probes/probes_main.c:114]: Fixed.

File: tools/probes/probes_main.c (hunk 2 above)
Comment (kv2019i) [reply tools/probes/probes_main.c]: Fixed.

**Comment by kv2019i** [comment]
Ready for review. Turns out we also need to add support for non-aligned sync words and payloads. Added this as a separate patch, retested and uploaded.

**Review by ujfalusi — APPROVED** [review]

**Comment by kv2019i** [comment]
The checkpatch issues around memcpy and function tab levels, are present in existing code, and I did not want to start refactoring surrounding code in this PR.

**Review by abonislawski — APPROVED** [review]
Good update to probe-app

**Review by jsarha — APPROVED** [review]
Looks good to.

**Review by lgirdwood — APPROVED** [review]

File: tools/probes/probes_main.c [hunk 3]
```diff
@@ -187,33 +187,44 @@ int validate_data_packet(struct probe_data_packet *data_packet)
...
+	*w_ptr = (uint8_t *)&packet->data;
 	return 0;
 }
 
+static bool sync_word_at(uint8_t *buf, size_t len)
+{
+	if (len < sizeof(uint32_t))
+		return false;
+
+	if (*((uint32_t *)buf) == PROBE_EXTRACT_SYNC_WORD)
```
Comment (lyakh) [inline tools/probes/probes_main.c:229]: `return *((uint32_t *)buf) == PROBE_EXTRACT_SYNC_WORD;` or even...

File: tools/probes/probes_main.c [hunk 4]
```diff
@@ -234,16 +245,17 @@ void parse_data(char *file_in)
 		fclose(fd_in);
 		exit(0);
 	}
-	memset(&data, 0, sizeof(uint32_t) * DATA_READ_LIMIT);
+	memset(&data, 0, DATA_READ_LIMIT);
```
Comment (lyakh) [inline tools/probes/probes_main.c:264]: love C. I guess here both `memset(&data...)` and `memset(data...)` would work and mean the same

File: tools/probes/probes_main.c [hunk 5]
```diff
@@ -234,16 +245,17 @@ void parse_data(char *file_in)
...
 	memset(&files, 0, sizeof(struct wave_files) * FILES_LIMIT);
 
 	/* data read loop to process DATA_READ_LIMIT bytes at each iteration */
 	do {
-		i = fread(&data, sizeof(uint32_t), DATA_READ_LIMIT, fd_in);
+		i = fread(&data, 1, DATA_READ_LIMIT, fd_in);
```
Comment (lyakh) [inline tools/probes/probes_main.c:269]: same here - and subjectively `fread(data,...)` seems clearer to me, but it's just me

File: tools/probes/probes_main.c [hunk 6]
```diff
@@ -100,8 +101,17 @@ int get_buffer_file_free(struct wave_files *files)
 	return -1;
 }
 
+bool is_audio_format(uint32_t format)
+{
+	if ((format & PROBE_MASK_FMT_TYPE) && (format & PROBE_MASK_AUDIO_FMT) == 0)
+		return true;
```
Comment (lyakh) [inline tools/probes/probes_main.c:107]: Is the above really correct? This is one of very rare cases where I get suspicious about missing parentheses - the above means `if ((format & PROBE_MASK_FMT_TYPE) != 0 && (format & PROBE_MASK_AUDIO_FMT) == 0)` - is that correct? If it is, then I'd use `== 0` and `!= 0` or just the boolean form consistently. And you can make it `return condition;`

File: tools/probes/probes_main.c [hunk 7]
```diff
@@ -153,6 +166,9 @@ void finalize_wave_files(struct wave_files *files)
 	/* and close all opened files */
 	/* check wave struct to understand the offsets */
 	for (i = 0; i < FILES_LIMIT; i++) {
+		if (!is_audio_format(files[i].fmt))
```
Comment (lyakh) [inline tools/probes/probes_main.c:169]: interesting, files are ordered so that audio files come first?

Comment (kv2019i) [reply tools/probes/probes_main.c:169]: @lyakh Yikes, no, this is a bug, I need to skip to next one, not break.

File: tools/probes/probes_main.c (hunk 6 above)
Comment (kv2019i) [reply tools/probes/probes_main.c:107]: The logic is correct, but I can update to make it easier to follow.

**Comment by kv2019i** [comment]
@lyakh found a bug in finalize wave header function. Turning to draft to block merge until bug is fixed.

**Comment by kv2019i** [comment]
Bug found by @lyakh fixed, pushed a new version. Ready for review and merge again.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**